### PR TITLE
fix(pg) harbor cluster pg status unknow

### PR DESCRIPTION
**Pull request by Ghostbaby**
harbor cluster pg status unknow

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: pkg/cluster/controllers/database/readiness.go [hunk 1]
```diff
@@ -43,17 +43,28 @@ const (
...
 
-	conn, _, err = p.GetInClusterDatabaseInfo(ctx)
+	conn, client, err = p.GetInClusterDatabaseInfo(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	defer client.Close(ctx)
+
+	if err := client.Ping(ctx); err != nil {
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:60]: We should not check the status with ping as the client in the operator process may not connect to psql instance deployed in another namespace.

Why do not use the status of psql CR?

@Ghostbaby

**Review by steven-zou — CHANGES_REQUESTED** [review]
Can we fix it ASAP?

File: pkg/cluster/controllers/database/readiness.go (hunk 1 above)
Comment (Ghostbaby) [reply pkg/cluster/controllers/database/readiness.go]: done

File: pkg/cluster/controllers/database/readiness.go [hunk 2]
```diff
@@ -49,11 +51,25 @@ func (p *PostgreSQLController) Readiness(ctx context.Context) (*lcm.CRStatus, er
 
 	name := p.HarborCluster.Name
 
-	conn, _, err = p.GetInClusterDatabaseInfo(ctx)
+	conn, err = p.GetInClusterDatabaseInfo()
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:54]: Why do we still need to get conn as we don't check health with that client anymore?

File: pkg/cluster/controllers/database/readiness.go [hunk 3]
```diff
@@ -49,11 +51,25 @@ func (p *PostgreSQLController) Readiness(ctx context.Context) (*lcm.CRStatus, er
...
 	if err != nil {
 		return nil, err
 	}
 
+	var pg api.Postgresql
+	if err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(p.ActualCR.UnstructuredContent(), &pg); err != nil {
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:61]: How do you make sure the actual CR is the latest version?

**Review by steven-zou — COMMENTED** [review]
comments left

File: pkg/cluster/controllers/database/readiness.go [hunk 4]
```diff
@@ -49,11 +50,21 @@ func (p *PostgreSQLController) Readiness(ctx context.Context) (*lcm.CRStatus, er
...
+		return nil, err
+	}
+
+	if pg.Status.PostgresClusterStatus != "Running" {
+		return nil, errors.New("fail to check pg")
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:65]:
```suggestion
		return nil, errors.New("pg cluster is not ready")
```

**Review by steven-zou — COMMENTED** [review]
comment left

**Review by heww — APPROVED** [review]

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**